### DISCUSSION TITLE
How to capture before launch event?

### DISCUSSION THREAD

**Discussion by ghost** [tauri-apps/tauri#3114] [Q&A] [answered]
Though I'm asking, I may already have this, maybe I just need a confirmation: I can do this through `main.rs` right? Before launching any window.

### Context
Take Windows' sticky notes for example. I have a "main" window (where the sticky notes are listed) and children (notes windows) launched from this window (with `WebViewWindow`).

I would store the last opened notes windows somewhere, and when the user launches a new instance from a fresh start (no previous window opened), it would launch these opened notes directly. It is possible doing this through main.rs, right?

**Comment by ghost** [comment]
The reason I'm asking this, is because Windows' Sticky Note app launches the main window **and then** opens the last opened sticky notes, and finally closes the "main" window. I don't want this.

**Comment by FabianLars** [comment, marked as answer]
The .setup hook on the Builder is probably your best bet.
Don't set any windows in tauri.conf and create them in setup.

**Reply by ghost** [reply]
Thank you, I'll try this.
